### PR TITLE
Send focus events even in ReadOnly mode

### LINKED ISSUE #13461: Windows Terminal Preview - Read-only pane pop-up on tab switch
### Windows Terminal version

Windows Terminal Preview Version: 1.15.1862.0

### Windows build number

10.0.19044.1766

### Other Software

Seems to be on anything I have open in terminal

### Steps to reproduce

Open 3 terminal tabs, PowerShell 7.2.5, Windows PowerShell 5.1, or command prompt. Set a keybind to toggle pane read-only mode (mine is set to ctrl-shift-\), then toggle one of the tabs to read-only.  Click on the 3 tabs in any order.  Pop-up stating "Read-only mode enabled" appears both switching to AND from that read-only tab to/from other tabs.
image
image

This doesn't happen in my currently installed non-preview version Windows Terminal Version: 1.13.11431.0 (I should probably update, but I don't use the non-preview version much)

### Expected Behavior

Personally, no popups at all, but if you have to, only when switching to the read-only tab.  I did not try with split panes, I don't use them much, though I like the feature in theory.

### Actual Behavior

Pop-up stating "Read-only mode enabled" appears both switching to AND from that read-only tab to/from other tabs.

I did discover that you can't set the settings tab to read-only, though I can live without that :)

### PR DESCRIPTION
**Pull request by zadjii-msft**
Does what it says on the tin. When we get focused, temporarily turn off readonly mode, as to not pop the dialog when the focus sequence is eventually sent to the connection. 

* closes #13461

### CODE REVIEW

**Review by DHowett — APPROVED** [review]

**Comment by DHowett** [comment]
Build go boom

**Review by DHowett — APPROVED** [review]

File: src/cascadia/TerminalControl/ControlCore.cpp [hunk 1]
```diff
@@ -1912,13 +1912,24 @@ namespace winrt::Microsoft::Terminal::Control::implementation
...
+    }
+
+    void ControlCore::_focusChanged(bool focused)
+    {
+        // GH#13461 - temporarily turn off read-only mode, send the focus event,
+        // then turn it back on. Even in focus mode, focus events are fine to
+        // send. We don't want to pop a warning every time the control is
+        // focused.
+        const auto previous = std::exchange(_isReadOnly, false);
+        const auto restore = wil::scope_exit([&]() { _isReadOnly = previous; });
```
Comment (PankajBhojwani) [inline src/cascadia/TerminalControl/ControlCore.cpp:1931]: nit: we don't seem to use `restore` anywhere, do we need it?

Comment (zadjii-msft) [reply src/cascadia/TerminalControl/ControlCore.cpp:1931]: yea, the way `wil::scope_exit` works, you _have_ to stash the result. When `restore` goes out of scope, then the body of the `scope_exit` will fire. If you don't stash it in a local, it'll destruct itself immediately, which in this case would automatically restore `_isReadOnly`

_(this is a mistake I've made in the past)_

**Review by PankajBhojwani — APPROVED** [review]

File: src/cascadia/TerminalControl/ControlCore.cpp (hunk 1 above)
Comment (PankajBhojwani) [reply src/cascadia/TerminalControl/ControlCore.cpp:1931]: Ah gotcha, thanks!

**Pull request merged**